src/wp-includes/block-template-utils.php

```diff
@@ -1163,14 +1163,40 @@ function get_block_templates( $query = array(), $template_type = 'wp_template' )
 	}
 
 	if ( ! isset( $query['wp_id'] ) ) {
+		$template_files_query = $query;
```

Comment:
Is this not important for the `get_block_templates` filter below? Before we were updating the same instance.

Reply:
Good call. I added an extra commit (cd07c085ba1c1c5e597b01596fbff44e385be4ae) to make sure `$query['slug__not_in']` stays consistent with what we had until now.